### PR TITLE
fix(telemetry): concurrency fixes

### PR DESCRIPTION
**Pull request by geclos**
In theory open telemetry spans are unique per trace but the reality is that nothing stops clients from sending spans with the same ID multiple times, and indeed they do. We handle this case by merging the spans on conflict where it is sensible to do so.

Also implemented some minor UI fixes and type fixes.

### CODE REVIEW

File: packages/core/drizzle/0113_quiet_wendigo.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE "latitude"."spans" DROP CONSTRAINT "spans_span_id_unique";--> statement-breakpoint
+ALTER TABLE "latitude"."spans" ADD CONSTRAINT "spans_span_id_trace_id_unique" UNIQUE("span_id","trace_id");
```
Comment (geclos) [inline packages/core/drizzle/0113_quiet_wendigo.sql:2]: span ids are only required to be unique per trace, not globally, so changing the index

File: packages/core/drizzle/0114_flippant_sunset_bain.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE "latitude"."spans" RENAME COLUMN "prompt_tokens" TO "input_tokens";--> statement-breakpoint
+ALTER TABLE "latitude"."spans" RENAME COLUMN "completion_tokens" TO "output_tokens";
```
Comment (geclos) [inline packages/core/drizzle/0114_flippant_sunset_bain.sql:2]: minor column renaming

File: packages/core/src/jobs/job-definitions/traces/processOtlpTracesJob.ts [hunk 3]
```diff
@@ -71,5 +71,5 @@ export const processOtlpTracesJob = async (
     traces: tracesToCreate,
     // @ts-expect-error - Fix when we fix types in compiler
     spans: processedSpans,
-  })
+  }).then((r) => r.unwrap())
```
Comment (geclos) [inline packages/core/src/jobs/job-definitions/traces/processOtlpTracesJob.ts:74]: i want this job to retry if the service fails for whatever reason so we unwrap

File: packages/core/src/services/traces/bulkCreateTracesAndSpans.ts [hunk 4]
```diff
@@ -215,60 +180,139 @@ async function createTraces(
...
+            WHEN ${spans.totalCostInMillicents} = 0 AND EXCLUDED.total_cost_in_millicents IS NOT NULL AND EXCLUDED.total_cost_in_millicents != 0
+            THEN EXCLUDED.total_cost_in_millicents
+            ELSE ${spans.totalCostInMillicents}
+          END
+        `,
+        documentUuid: sql`COALESCE(${spans.documentUuid}, EXCLUDED.document_uuid)`,
+        distinctId: sql`COALESCE(${spans.distinctId}, EXCLUDED.distinct_id)`,
+        commitUuid: sql`COALESCE(${spans.commitUuid}, EXCLUDED.commit_uuid)`,
+        updatedAt: new Date(),
+      },
```
Comment (geclos) [inline packages/core/src/services/traces/bulkCreateTracesAndSpans.ts:315]: the meat is here, basically merging duplicated spans via "on conflict" postgres handler and a bunch of sql rules

File: packages/web-ui/src/ds/molecules/Chat/Message/index.tsx [hunk 5]
```diff
@@ -65,7 +65,7 @@ export function MessageItem({
           className='flex-shrink-0 bg-muted w-1 rounded-lg cursor-pointer hover:bg-primary transition-colors'
           onClick={() => setCollapseMessage(!collapsedMessage)}
         />
-        <div className='flex flex-1 flex-col gap-1'>
+        <div className='flex flex-grow flex-col gap-1 overflow-x-auto'>
```
Comment (andresgutgon) [inline packages/web-ui/src/ds/molecules/Chat/Message/index.tsx:68]: Be aware that this component is also used in shared prompts. Maybe check if it looks ok

File: packages/core/src/services/traces/bulkCreateTracesAndSpans.ts (hunk 4 above)
Comment (andresgutgon) [reply packages/core/src/services/traces/bulkCreateTracesAndSpans.ts:315]: A lot of meat here 🤯

**Review by andresgutgon — DISMISSED** [review]

File: packages/web-ui/src/ds/molecules/Chat/Message/index.tsx (hunk 5 above)
Comment (geclos) [reply packages/web-ui/src/ds/molecules/Chat/Message/index.tsx:68]: reviewed and it looks good

**Pull request merged**